### PR TITLE
readpool: fix missing metric `tikv_yatp_task_poll_duration`

### LINKED ISSUE #14424: The histogram `tikv_yatp_task_poll_duration` is missing
## Bug Report

After resource control became default, some metrics are missing.

### What version of TiKV are you using?

I'm running nightly TiKV.

```
TiKV 
Release Version:   6.7.0-alpha
Edition:           Community
Git Commit Hash:   21ef364077bdefbd306a26019879c75a3687a27b
Git Commit Branch: heads/refs/tags/v6.7.0-alpha
UTC Build Time:    2023-03-16 14:35:00
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```

### Steps to reproduce

Run a cluster.

```sh
tiup playground nightly --tiflash=0 --without-monitor
```

Run sysbench read only

```sh
sysbench --config-file=config oltp_read_only --tables=1 --table-size=100000 --auto_inc=false prepare
sysbench --config-file=config oltp_read_only --tables=1 --table-size=100000 --auto_inc=false run
```

Check the metrics from http://127.0.0.1:20180/metrics

### What did you expect?

`tikv_yatp_task_poll_duration` is recorded.

### What did happened?

`tikv_yatp_task_poll_duration` is zero.

```
tikv_yatp_task_poll_duration_sum{level="0",name="unified-read-pool"} 0
tikv_yatp_task_poll_duration_count{level="0",name="unified-read-pool"} 0
...
tikv_yatp_task_poll_duration_sum{level="1",name="unified-read-pool"} 0
tikv_yatp_task_poll_duration_count{level="1",name="unified-read-pool"} 0
...
tikv_yatp_task_poll_duration_sum{level="2",name="unified-read-pool"} 0
tikv_yatp_task_poll_duration_count{level="2",name="unified-read-pool"} 0
```

### PR DESCRIPTION
**Pull request by you06**
### What is changed and how it works?

Issue Number: Close #14424

What's Changed:

```commit-message
Fix the missing metric `tikv_yatp_task_poll_duration` by upgrading yatp.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by you06** [comment]
The regression test failed before upgrading yatp.

image

**Comment by cfzjywxk** [comment]
@glorv @nolouch @BusyJay 
PTAL it's needed for the release-7.0, thx.

File: src/read_pool.rs [hunk 1]
```diff
@@ -419,7 +419,7 @@ pub fn build_yatp_read_pool<E: Engine, R: FlowStatsReporter>(
     engine: E,
     resource_ctl: Option<Arc<ResourceController>>,
     cleanup_method: CleanupMethod,
-) -> ReadPool {
+) -> (ReadPool, String) {
```
Comment (nolouch) [inline src/read_pool.rs:422]: seem the name `String` only used in test? Do we really need it?

**Review by nolouch — COMMENTED** [review]
rest lgtm

File: src/read_pool.rs (hunk 1 above)
Comment (glorv) [reply src/read_pool.rs:422]: +1. What about add a test method that accepts the name as input parameter.

Comment (you06) [reply src/read_pool.rs]: I made a wrap for common usage.

**Review by glorv — APPROVED** [review]
LGTM

File: src/read_pool.rs (hunk 1 above)
Comment (glorv) [reply src/read_pool.rs]: NIT: I think another way to do expose the name is to put a name field only for test in the ReadPool, but the current implementation is ok to me.

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**